src/AnnotationDialog.js

```diff
@@ -45,10 +44,11 @@ class AnnotationDialog extends EventEmitter {
     constructor(data) {
         super();
 
+        this.annotations = [];
+
         this.annotatedElement = data.annotatedElement;
         this.container = data.container;
         this.location = data.location;
-        this.hasAnnotations = Object.keys(data.annotations).length > 0;
```

Comment:
Turn this into a method? I think it's cleaner than checking annotations.length > 0 in a bunch of places.